src/lambdaisland/deep_diff2/diff_impl.cljc

```diff
@@ -109,26 +110,19 @@
    (remove #(contains? exp %) act)))
 
 (defn diff-map [exp act]
-  (first
-   (let [exp-ks (keys exp)
-         act-ks (keys act)
-         [del ins] (del+ins exp-ks act-ks)]
-     (reduce
-      (fn [[m idx] k]
-        [(cond-> m
-           (contains? del idx)
-           (assoc (->Deletion k) (get exp k))
-
-           (not (contains? del idx))
-           (assoc k (diff (get exp k) (get act k)))
-
-           (contains? ins idx)
-           (into (map (juxt ->Insertion (partial get act))) (get ins idx)))
-         (inc idx)])
-      [(if (contains? ins -1)
-         (into {} (map (juxt ->Insertion (partial get act))) (get ins -1))
-         {}) 0]
-      exp-ks))))
+  (let [exp-ks (set (keys exp))
+        act-ks (set (keys act))]
+    (reduce
+      (fn [m k]
+        (case [(contains? exp-ks k) (contains? act-ks k)]
+          [true false]
+          (assoc m (->Deletion k) (get exp k))
+          [false true]
+          (assoc m (->Insertion k) (get act k))
+          [true true]
+          (assoc m k (diff (get exp k) (get act k)))))
```

Comment:
It might be good to explain why the `[false false]` case isn't needed (every key is in at least one of the two maps, since we're reducing over the union of their keys).